### ISSUE TITLE
Mobile menu not showing

### ISSUE THREAD

**Issue by dcman007** [dupkey/bs4navwalker#18]
I'm using the navwalker for a WordPress project. Everything is fine, with one exception. When clicking on the mobile hamburger button the drop-down menu doesn't show. The code is being fired correctly because I can see it's adding the class 'show' to the navbar-collapse class. When inspecting the css navbar-collapse {display:none} is still active, switching this off manually in the code shows the drop-down.

I've pasted the code in below - appreciate any pointers as to what is going wrong with this.

*** Code from my functions File ***

/**Enqueue scripts and styles. */

```
function quizzes_scripts() {
	wp_enqueue_style( 'quizzes-style', get_stylesheet_uri() );

	if ( is_singular() && comments_open() && get_option( 'thread_comments' ) ) {
		wp_enqueue_script( 'comment-reply' );
	}
}
add_action( 'wp_enqueue_scripts', 'quizzes_scripts' );

/** add in bootstrap scripts */
function bootstrap_scripts() { 
	wp_enqueue_style('bootstrap-styles', get_template_directory_uri() . '/css/bootstrap.min.css', array (), '4.0', 'all');
	wp_enqueue_style( 'bootflat-styles', get_template_directory_uri() . '/css/bootflat.min.css', array(), '1.0', 'all' );
    wp_enqueue_style( 'bootstrap-style', get_stylesheet_uri() );
    //wp_enqueue_script( 'bootstrap-js', get_template_directory_uri() . '/js/bootstrap.min.js', array(), '4.0', true );
}
add_action( 'wp_enqueue_scripts', 'bootstrap_scripts' );

// Custom Navigation Walker
require_once('bs4navwalker.php');

/** register menus */
function register_my_menus() {
  register_nav_menus(
  	array(
  	'main-menu' => __( 'Main Menu' ),
	'footer-menu' => __('Footer Menu')  
  	)
  );
}
add_action( 'init', 'register_my_menus' );
```

*** code in my template file to generate the menu ***
```
<div class="container-fluid bg-lilac remove-padding">
			<div class="container">
				<div class="col-sm">
					<nav class="navbar navbar-expand-lg navbar-light">
						<button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#bs4navbar" aria-controls="bs4navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>
						<div class="collapse navbar-collapse justify-content-center" id="navbar-content">
							<?php
							// Menu Location
							/* wp_nav_menu( array(
								'menu' => 'Main Menu',
								'theme_location' => 'Primary Menu',
								'depth' => 2,
								'container' => 'div',
								'container_class' => 'collapse navbar-collapse',
								'container_id' => 'bs-example-navbar-collapse-1',
								'menu_class' => 'nav navbar-nav navbar-center',
								'fallback_cb' => 'wp_bootstrap_navwalker::fallback',
								'walker' => new wp_bootstrap_navwalker() ) );*/
							
							
							
								wp_nav_menu([
     								'menu'            => 'Main Menu',
     								'theme_location'  => 'Primary Menu',
     								'container'       => 'div',
     								'container_id'    => 'bs4navbar',
     								'container_class' => 'collapse navbar-collapse',
									'menu_id'         => false,
     								'menu_class'      => 'navbar-nav mr-auto',
     								'depth'           => 2,
     								'fallback_cb'     => 'bs4navwalker::fallback',
									'walker'          => new bs4navwalker()
   								]);
 
							
							?>
						</div><!-- ./navbar-collapse -->
					</nav><!-- ./nav -->
				</div><!-- ./columns -->
			</div><!-- ./container -->
			</div>
			<!-- ./container-fluid-->
```

**Comment by dupkey** [comment]
You've commented out bootstrap.min.js -

`//wp_enqueue_script( 'bootstrap-js', get_template_directory_uri() . '/js/bootstrap.min.js', array(), '4.0', true );`

**Comment by dcman007** [comment]
Sorry I didn't include the code from the footer.php file - the bootstrap.js include is in there.

```
    <!-- jQuery first, then Popper.js, then Bootstrap JS -->
   <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>

    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
```

**Comment by dupkey** [comment]
The only other thing I can think of, based on the code provided, is if bootflat has some conflicting css.

If you have a link to the site I can take a look.

**Comment by dcman007** [comment]
Thanks for offering to help, it's very much appreciated. The website isn't actually live yet, we have only just begun development on it. We had to add an entry in the hosts file on our local development machine to point the domain at the IP of the dev server.

Add the following to your hosts file:

77.72.0.94     instantquizzes.co.uk
77.72.0.94     www.instantquizzes.co.uk

You should be able to see the dev site then.

I know I am missing something obvious, but I can't for the life of me see what that is. Been stuck on this for a few days now.

**Comment by dupkey** [comment]
I've managed to get it working. A couple things going on here:

On the line following the comment  you have included the script bootstrap.min.js however that is the wrong place for it. It needs to go after jquery and popper.

You do have the script included just before the closing </body> tag, however it is commented out. If you uncomment that script and remove the proceeding script your menu should work.

I tested it in Chrome dev tools by appending bootstrap.min.js to the body and everything worked. Example below in case interested. Let me know if that does the trick.

var script = document.createElement('script');
script.type = 'text/javascript';
script.src = 'https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js';
document.head.appendChild(script);
